lib/logstash/inputs/beats.rb

```diff
@@ -207,9 +210,229 @@ def invoke(connection, &block)
   # this thread take the element from one queue into another one.
   def start_buffer_broker
     @threadpool.post do
-      while !stop?
-        @output_queue << @buffered_queue.pop_no_timeout
+      LogStash::Util.set_thread_name("[beats-input]-buffered-queue-broker")
+
+      begin
+        while !stop?
+          @output_queue << @buffered_queue.take
+        end
+      rescue => e
+        # If we are shutting down without waiting the queue to unblock
+        # we will get an `InterruptionException` in that context we will not log it.
+        @logger.error("Beats input: bufferered queue broker exception", :exception => e) unless stop?
       end
     end
   end
+
+  # Handle the data coming from a connection
+  # Decide which Process should be used to decode the data coming
+  # from the beat library.
+  #
+  # - Should we use a codec on specific field?
+  # - Should we just take the raw content of the parsed json frame
+  class ConnectionHandler
```

Comment:
I would +1 this but the way our plugins are structured doesn't make it easy, theses classes should be inside a "LogStash::Inputs::Beats" namespace which we can't do because we have a class named beats. I wish the plugin structure wasn't directed by the logstash main structure.

Would you be OK to group theses classes inside a `BeatsSupport` namespace?